src/cmd/run.go

```diff
@@ -0,0 +1,335 @@
+		}
+		containerName = args[0]
+		commands = args[1:]
+	}
+
+	containerName, imageName, _ = utils.UpdateContainerAndImageNames(containerName, imageName, runFlags.releaseVersion)
+
+	logrus.Debugf("Container: '%s' Image: '%s'", containerName, imageName)
+
+	logrus.Infof("Checking if container '%s' exists", containerName)
+	if !podman.ContainerExists(containerName) {
+		if !runFlags.pedantic {
+			logrus.Errorf("Container '%s' not found", containerName)
+			containers, err := GetContainers()
+			if err != nil {
+				logrus.Fatal("Error while fetching containers")
+			}
+
+			containersCount := len(containers)
+			logrus.Infof("Found %d containers", containersCount)
+
+			if containersCount == 0 {
+				createContainer := false
+
+				if rootFlags.assumeyes {
+					runFlags.promptForCreate = false
+					createContainer = true
+				}
+
+				if runFlags.promptForCreate {
+					var response string
+					for true {
+						fmt.Printf("No toolbox containers found. Create now? [y/N]: ")
+						fmt.Scanf("%s", &response)
+						response = strings.ToLower(response)
+
+						if response == "y" || response == "yes" {
+							createContainer = true
+						}
+
+						break
+					}
+				}
+
+				if !createContainer {
+					fmt.Println("A container can be created later with the 'create' command")
+					return nil
+				}
+
+				create([]string{containerName})
+			} else if containersCount == 1 {
+				containerName = fmt.Sprint(containers[0]["Names"])
+				logrus.Infof("Entering container %s instead", containerName)
+			} else {
+				logrus.Fatal("Create a new container with 'toolbox create' or 'toolbox enter' an already existing container")
+			}
+		} else {
+			logrus.Fatalf("Container '%s' not found", containerName)
+		}
+	}
+
+	// Prepare Flatpak session-helper
+	conn, err := dbus.SessionBus()
+	if err != nil {
+		logrus.Error("Failed to connect to Session Bus")
+	}
+
+	logrus.Info("Calling org.freedesktop.Flatpak.SessionHelper.RequestSession")
+	SessionHelper := conn.Object("org.freedesktop.Flatpak", "/org/freedesktop/Flatpak/SessionHelper")
+	call := SessionHelper.Call("org.freedesktop.Flatpak.SessionHelper.RequestSession", 0)
+	if call.Err != nil {
+		logrus.Debug(call.Err)
+		logrus.Fatal("Failed to call org.freedesktop.Flatpak.SessionHelper.RequestSession")
+	}
+
+	logrus.Infof("Starting container '%s'", containerName)
+	err = containerStart(containerName)
+	if err != nil {
+		logrus.Fatal(err)
+	}
+
+	logrus.Infof("Inspecting container '%s'", containerName)
+	containerInfo, err := podman.PodmanInspect("container", containerName)
+	if err != nil {
+		logrus.Fatal(err)
+	}
+
+	containerConfig := containerInfo["Config"].(map[string]interface{})
+	containerState := containerInfo["State"].(map[string]interface{})
+	containerEntryPoint := containerConfig["Cmd"].([]interface{})[0]
+
+	logrus.Infof("Entry point of container '%s' is '%s'", containerName, containerEntryPoint)
+
+	if containerEntryPoint == "toolbox" {
```

Comment:
We should simply reject containers that don't have `toolbox` itself as their entry point. Such containers have been marked as deprecated since the 0.0.17 release in November 2019.